### ISSUE TITLE
Pure HTML Loader via HTMLLayoutData causing a loading error

### ISSUE THREAD

**Issue by tsukhu** [single-spa/single-spa-layout#47]
When we try to configure the loader ui via the `contructRoutes` using a pure HTML loader it throws the following error

```
10:03:44.584 Navigated to http://localhost:9000/
10:03:45.306 load.js:96 single-spa minified message #37: See https://single-spa.js.org/error/?code=37&arg=application&arg=@react-mf/navbar&arg=[null,{"__esModule":true}] (2) [null, Module]
(anonymous) @ single-spa.min.js:2
Promise.then (async)
(anonymous) @ single-spa.min.js:2
Promise.then (async)
(anonymous) @ single-spa.min.js:2
Promise.then (async)
X @ single-spa.min.js:2
(anonymous) @ single-spa.min.js:2
Promise.then (async)
f @ single-spa.min.js:2
Lt @ single-spa.min.js:2
Pt @ single-spa.min.js:2
./src/root-config.js @ root-config.js:29
__webpack_require__ @ bootstrap:18
0 @ react-mf-root-config.js:9977
__webpack_require__ @ bootstrap:18
(anonymous) @ bootstrap:82
execute @ bootstrap:83
i @ system.min.js:4
e @ system.min.js:4
(anonymous) @ system.min.js:4
Promise.then (async)
(anonymous) @ system.min.js:4
(anonymous) @ system.min.js:4
Promise.then (async)
w.import @ system.min.js:4
o.import @ system.min.js:4
(anonymous) @ (index):46
10:03:45.307 app-errors.js:75 single-spa minified message #30: See https://single-spa.js.org/error/?code=30&arg=LOADING_SOURCE_CODE&arg=@react-mf/navbar
l @ single-spa.min.js:2
a @ single-spa.min.js:2
(anonymous) @ single-spa.min.js:2
Promise.then (async)
(anonymous) @ single-spa.min.js:2
Promise.then (async)
(anonymous) @ single-spa.min.js:2
Promise.then (async)
X @ single-spa.min.js:2
(anonymous) @ single-spa.min.js:2
Promise.then (async)
f @ single-spa.min.js:2
Lt @ single-spa.min.js:2
Pt @ single-spa.min.js:2
./src/root-config.js @ root-config.js:29
__webpack_require__ @ bootstrap:18
0 @ react-mf-root-config.js:9977
__webpack_require__ @ bootstrap:18
(anonymous) @ bootstrap:82
execute @ bootstrap:83
i @ system.min.js:4
e @ system.min.js:4
(anonymous) @ system.min.js:4
Promise.then (async)
(anonymous) @ system.min.js:4
(anonymous) @ system.min.js:4
Promise.then (async)
w.import @ system.min.js:4
o.import @ system.min.js:4
(anonymous) @ (index):46
```

**Steps to reproduce**

1. Clone the repo https://github.com/tsukhu/root-config
2. Run `yarn`
3. Run `yarn start`

The code in question is in the root-config.js
```
const data = {
  loaders: {
    navLoader: `<div>Loading...</div>`,
  },
};

const routes = constructRoutes(
  document.querySelector('#single-spa-layout'),
  data
);
```
 and the index.ejs file has reference to the `navLoader`
```
  <template id="single-spa-layout">
    <single-spa-router>
      <nav class="topnav">
        <application name="@react-mf/navbar" loader="navLoader"></application>
      </nav>
  ...
```

Note: If you replace the data object with a blank loader then it works

```
const data = {
  loaders: {
    navLoader: ``,
  },
};
```

**Comment by joeldenning** [comment]
Thanks for the clear description and repo that demonstrates the issue. I confirmed that it's currently broken and am putting together a fix.

**Comment by joeldenning** [comment]
Fix for this is at #48

**Comment by tsukhu** [comment]
While the error now has gone , but still there may be an issue with the loader not working as expected.

To demonstrate I have updated the same repo https://github.com/tsukhu/root-config with a bright yellow loader with a very large font :-) 

```
const data = {
  loaders: {
    navLoader: `<div style="display:block;background-color:yellow;color:black;padding:20px;font-size:100px;">Loading...</div>`,
  },
};
```
Since the navbar loads very fast you can simulate the issue like this

1) Use the same repo  https://github.com/tsukhu/root-config . This has been updated with the latest version of the layout library
2) run `yarn`
3) run `yarn start`
4) load `localhost:9000` in the browser

You will see a yellow flash before the navbar loads

Now change the import map dynamically using the importmaps override

![step 1](https://user-images.githubusercontent.com/846306/84733168-2011bd00-afbb-11ea-8392-1bd04cf9cfe5.PNG)

5) refresh the browser
6) You will see the yellow line now .. but there is no content
on inspecting closely we see that the `Loading...` text is missing

![Actual](https://user-images.githubusercontent.com/846306/84733252-5c451d80-afbb-11ea-9c09-3d6ba7e48f04.PNG)

What we should be getting is something like this where the `Loading...` text is present (I have manually edited the html in the browser to show this)

![expected](https://user-images.githubusercontent.com/846306/84733266-66671c00-afbb-11ea-95de-392552cd4304.PNG)

which also brings me to another possible improvement / suggestion on handling of failure to load an application we need a way to show a relavent message in placeholder like `currently this feature is not available` or something

**Comment by joeldenning** [comment]
I've confirmed that the Loading text is missing, and have a fix for it. Reopening this while that PR is pending.

**Comment by joeldenning** [comment]
Another thing I discovered while testing this is that the loader wasn't showing for as long as it should - it would only flash momentarily due to a bug. I have fixed both of the issues in #50

**Comment by joeldenning** [comment]
Fix released in https://github.com/single-spa/single-spa-layout/releases/tag/v1.0.0-beta.4
